pandas/tests/series/test_operators.py

```diff
@@ -595,77 +595,103 @@ def test_divide_decimal(self):
 
         assert_series_equal(expected, s)
 
-    def test_div(self):
+    def test_i8_ser_div_i8_ser(self):
+        # no longer do integer div for any ops, but deal with the 0's
+        first = Series([3, 4, 5, 8], name='first')
+        second = Series([0, 0, 0, 3], name='second')
+
         with np.errstate(all='ignore'):
-            # no longer do integer div for any ops, but deal with the 0's
-            p = DataFrame({'first': [3, 4, 5, 8], 'second': [0, 0, 0, 3]})
-            result = p['first'] / p['second']
-            expected = Series(
-                p['first'].values.astype(float) / p['second'].values,
-                dtype='float64')
-            expected.iloc[0:3] = np.inf
-            assert_series_equal(result, expected)
+            expected = Series(first.values.astype(np.float64) / second.values,
+                              dtype='float64')
+        expected.iloc[0:3] = np.inf
 
-            result = p['first'] / 0
-            expected = Series(np.inf, index=p.index, name='first')
-            assert_series_equal(result, expected)
+        result = first / second
+        assert_series_equal(result, expected)
 
-            p = p.astype('float64')
-            result = p['first'] / p['second']
-            expected = Series(p['first'].values / p['second'].values)
-            assert_series_equal(result, expected)
+    def test_f8_ser_div_f8_ser(self):
+        # no longer do integer div for any ops, but deal with the 0's
+        first = Series([3, 4, 5, 8], name='first', dtype='float64')
+        second = Series([0, 0, 0, 3], name='second', dtype='float64')
 
-            p = DataFrame({'first': [3, 4, 5, 8], 'second': [1, 1, 1, 1]})
-            result = p['first'] / p['second']
-            assert_series_equal(result, p['first'].astype('float64'),
-                                check_names=False)
-            assert result.name is None
-            assert not result.equals(p['second'] / p['first'])
-
-            # inf signing
-            s = Series([np.nan, 1., -1.])
-            result = s / 0
-            expected = Series([np.nan, np.inf, -np.inf])
-            assert_series_equal(result, expected)
+        result = first / second
+        with np.errstate(all='ignore'):
+            expected = Series(first.values / second.values)
+        assert_series_equal(result, expected)
 
-            # float/integer issue
-            # GH 7785
-            p = DataFrame({'first': (1, 0), 'second': (-0.01, -0.02)})
-            expected = Series([-0.01, -np.inf])
+    def test_ser_div_ser_name_propagation(self):
```

Comment:
why is this a separate test and not in the _div_ test?

Reply:
Not really sure what the original motivation was here.  TBH I was just guessing that the point was name propagation, since it wasn't clear what else it could be for.